internal/test/builder/builders.go

```diff
@@ -1287,7 +1548,7 @@ func (m *MachineSetBuilder) WithBootstrapTemplate(ref *unstructured.Unstructured
 	return m
 }
 
-// WithInfrastructureTemplate adds the passed unstructured object to the MachineSetBuilder as an infrastructureMachineTemplate.
+// WithInfrastructureTemplate adds the passed unstructured object to the MachineSetBuilder as an infrastructureMachinePoolTemplate.
```

Comment:
This is for MachineSets :-)

```suggestion
// WithInfrastructureTemplate adds the passed unstructured object to the MachineSetBuilder as an infrastructureMachineTemplate.
```